NossoMercadoLivre/Models/Entities/User.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using Dapper.Contrib.Extensions;
+using System.ComponentModel.DataAnnotations;
+
+namespace NossoMercadoLivre.Models.Entities
+{
+    [Table("Users")]
+    public class User
```

Comment:
*Usamos tudo que conhecemos que está pronto. Só fazemos código do zero se for estritamente necessário.*

Um usuário tem instante de criação, login e senha obrigatórios. Enquanto que o instante não é definido via parâmetro, o login e senha são. E coisas que são parametrizáveis e obrigatórias entram pelo construtor. 

Neste momento não precisamos possibilitar nenhuma alteração parcial de objeto também, então não precisamos de setter .Também não tratamos do instante de atualização, então pode remover também :). 

Sobre o id ser gerado no construtor, sem problemas. Essa é a geração mais comum? Nao rola configurar um identity ou algo assim?

Reply:
Eu gosto de gerar o Id como Guid mesmo, mas poderia utilizar o int, ou um long e deixar gerar um Identity do banco.